packages/react-paypal-js/src/v6/hooks/useCardFieldsOneTimePaymentSession.test.ts

```diff
@@ -0,0 +1,221 @@
+import { CARD_FIELDS_SESSION_TYPES } from "../components/CardFieldsProvider";
+import { toError } from "../utils";
+
+import type {
+    CardFieldsOneTimePaymentSession,
+    OneTimePaymentFlowResponse,
+} from "../types";
+
+jest.mock("./useCardFields");
+
+const mockUseCardFieldsSession = useCardFieldsSession as jest.MockedFunction<
+    typeof useCardFieldsSession
+>;
+
+const mockOrderId = "test-order-id";
+
+const mockOptions = { billingAddress: { postalCode: "12345" } };
+
+const mockSuccessfulSubmitResponse: OneTimePaymentFlowResponse = {
+    data: {
+        orderId: mockOrderId,
+    },
+    state: "succeeded",
+};
+
+// Mock Factories
+const createMockCardFieldsOneTimePaymentSession = (
+    overrides?: Partial<CardFieldsOneTimePaymentSession>,
+): CardFieldsOneTimePaymentSession => ({
+    submit: jest.fn().mockResolvedValue(undefined),
+    on: jest.fn(),
+    update: jest.fn(),
+    createCardFieldsComponent: jest.fn(),
+    ...overrides,
+});
+
+describe("useCardFieldsOneTimePaymentSession", () => {
+    let mockCardFieldsOneTimePaymentSession: CardFieldsOneTimePaymentSession;
+
+    beforeEach(() => {
+        mockCardFieldsOneTimePaymentSession =
+            createMockCardFieldsOneTimePaymentSession({
+                submit: jest
+                    .fn()
+                    .mockResolvedValue(mockSuccessfulSubmitResponse),
+            });
+
+        mockUseCardFieldsSession.mockReturnValue({
+            cardFieldsSession: mockCardFieldsOneTimePaymentSession,
+            cardFieldsSessionType: CARD_FIELDS_SESSION_TYPES.ONE_TIME_PAYMENT,
+        });
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    test("should return the expected object structure", () => {
+        const { result } = renderHook(() =>
+            useCardFieldsOneTimePaymentSession(),
+        );
+
+        expect(result.current).toHaveProperty("submit");
+        expect(result.current).toHaveProperty("error");
+        expect(result.current).toHaveProperty("submitResponse");
+    });
+
+    describe("submit", () => {
+        test("should provide a submit function", () => {
+            const { result } = renderHook(() =>
+                useCardFieldsOneTimePaymentSession(),
+            );
+
+            expect(typeof result.current.submit).toBe("function");
+        });
+
+        test("should call the session's submit method with correct arguments", async () => {
+            const { result } = renderHook(() =>
+                useCardFieldsOneTimePaymentSession(),
+            );
+
+            await act(async () => {
+                await result.current.submit(mockOrderId, mockOptions);
+            });
+
+            expect(
+                mockCardFieldsOneTimePaymentSession.submit,
+            ).toHaveBeenCalledWith(mockOrderId, mockOptions);
+        });
+
+        test("should update submitResponse with the session's submit response", async () => {
```

Comment:
I feel like this test and the previous one could be rolled up into one test, for example: "`submit` should to be called with arguments X and return Y".